### ISSUE TITLE
Broken TP-LINK Jetsream after #12990

### ISSUE THREAD

**Issue by Npeca75** [librenms/librenms#13050]
Hi 
after @jbronn  PR 
https://github.com/librenms/librenms/pull/12990
which added DDM support fort TP-Link Jetstream, the switch SNMP engine is broken

`#### Load disco module processors ####

Caching data: os mempools processors 
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'COMMUNITY' '-OteQUs' '-Ih' '-m' 'TPLINK-SYSINFO-MIB:TPLINK-SYSMONITOR-MIB:TPLINK-DDMSTATUS-MIB:TPLINK-DDMBIASCURTHRESHOLD-MIB:TPLINK-DDMRXPOWTHRESHOLD-MIB:TPLINK-DDMTXPOWTHRESHOLD-MIB:TPLINK-DDMTEMPTHRESHOLD-MIB:TPLINK-DDMVOLTHRESHOLD-MIB:TPLINK-POWER-OVER-ETHERNET-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/tplink' 'udp:HOSTNAME:161' 'tpSysMonitorCpu5Seconds']

tpSysMonitorCpu5Seconds.1 = 64  

sensors 
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'COMMUNITY' '-OteQUs' '-Ih' '-m' 'TPLINK-SYSINFO-MIB:TPLINK-SYSMONITOR-MIB:TPLINK-DDMSTATUS-MIB:TPLINK-DDMBIASCURTHRESHOLD-MIB:TPLINK-DDMRXPOWTHRESHOLD-MIB:TPLINK-DDMTXPOWTHRESHOLD-MIB:TPLINK-DDMTEMPTHRESHOLD-MIB:TPLINK-DDMVOLTHRESHOLD-MIB:TPLINK-POWER-OVER-ETHERNET-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/tplink' 'udp:HOSTNAME:161' 'ddmBiasCurThresholdLowAlarm']

Exitcode: 1  
Timeout: No Response from udp:169.254.102.39:161  `

SNMP engine stay broken until switch reboot

HardwareT2600G-18TS 2.0
Operating SystemTP-Link JetStream 2.0.3 Build 20190516 Rel.33426(s)
Serial217B719000242

**Comment by PipoCanaja** [comment]
Hi @Npeca75 
Have you tried upgrading the switch to the latest firmware ? If yes, you should open a case at the vendor, this is a switch bug, not a LibreNMS bug. 
Bye

**Comment by jbronn** [comment]
@Npeca75 I tested #12990 on a T2600G-28TS 3.0 running its latest firmware (`3.0.3 Build 20181101 Rel.42543(s)`).  These are older switches with anemic CPUs, which is why I think TP-Link doesn't support them with their Omada SDN.  I set `nobulk` in my configuration, but didn't want to impose this globally because of the age of my hardware (e.g., the changes had no perf impact on my newer SG2210P v3.0).  

tl;dr set this in your `config.php` for improved performance with this older hardware:
```php
// Streamline performance for older Jetstream hardware versions like T2600G-28TS <= 3.0.
$config['os']['jetstream']['nobulk'] = true;
$config['os']['jetstream']['discovery_modules']['hr-device'] = false;
$config['os']['jetstream']['discovery_modules']['services'] = false;
$config['os']['jetstream']['discovery_modules']['storage'] = false;
$config['os']['jetstream']['discovery_modules']['ucd-diskio'] = false;
$config['os']['jetstream']['discovery_modules']['wireless'] = false;
$config['os']['jetstream']['poller_modules']['hr-mib'] = false;
$config['os']['jetstream']['poller_modules']['ipmi'] = false;
$config['os']['jetstream']['poller_modules']['ipSystemStats'] = false;
$config['os']['jetstream']['poller_modules']['ucd-diskio'] = false;
$config['os']['jetstream']['poller_modules']['ucd-mib'] = false;
$config['os']['jetstream']['poller_modules']['unix-agent'] = false;
$config['os']['jetstream']['poller_modules']['wireless'] = false;
```

**Comment by Npeca75** [comment]
hi @jbronn 

> ```
> $config['os']['jetstream']['nobulk'] = true
> ```

this line fixed the problem
looks like there is much to learn about LNMS every day ...

thank you

**Comment by Npeca75** [comment]
hi @PipoCanaja 

> Have you tried upgrading the switch to the latest firmware ? If yes, you should open a case at the vendor, this is a switch bug, 

no, the 20190516 Rel.33426(s) is last FW
when i bought this switch, it looked like a good piece of HW
stable operation more than 1 year, console port, good features ...
but there is no support after 2019

**Comment by Npeca75** [comment]
hi @jbronn 
as you was original author of tplink DDM PR
i like to ask you, do you have any objection if i put these OIDs in jetstream YAML and send PR?
there is newly introduced 'no_bulk' option in YAML to disable bulkwalk only on specific OIDs
this way, we could avoid messing with config.php , and take off 'nobulk' option from whole jetstream OS

from my point of view, any read from DDM* is already slow so leaving only DDM* to plain walk will not avoid performance

BR Peca

$config['os']['jetstream']['oids']['no_bulk'][] = 'ddmBiasCurThresholdLowAlarm';
$config['os']['jetstream']['oids']['no_bulk'][] = 'ddmBiasCurThresholdLowWarn';
$config['os']['jetstream']['oids']['no_bulk'][] = 'ddmBiasCurThresholdHighAlarm';
$config['os']['jetstream']['oids']['no_bulk'][] = 'ddmBiasCurThresholdHighWarn';
$config['os']['jetstream']['oids']['no_bulk'][] = 'ddmTempThresholdLowAlarm';
$config['os']['jetstream']['oids']['no_bulk'][] = 'ddmTempThresholdLowWarn';
$config['os']['jetstream']['oids']['no_bulk'][] = 'ddmTempThresholdHighAlarm';
$config['os']['jetstream']['oids']['no_bulk'][] = 'ddmTempThresholdHighWarn';
$config['os']['jetstream']['oids']['no_bulk'][] = 'ddmVolThresholdLowAlarm';
$config['os']['jetstream']['oids']['no_bulk'][] = 'ddmVolThresholdLowWarn';
$config['os']['jetstream']['oids']['no_bulk'][] = 'ddmVolThresholdHighAlarm';
$config['os']['jetstream']['oids']['no_bulk'][] = 'ddmVolThresholdHighWarn';
$config['os']['jetstream']['oids']['no_bulk'][] = 'ddmRxPowThresholdLowAlarm';
$config['os']['jetstream']['oids']['no_bulk'][] = 'ddmRxPowThresholdLowWarn';
$config['os']['jetstream']['oids']['no_bulk'][] = 'ddmRxPowThresholdHighAlarm';
$config['os']['jetstream']['oids']['no_bulk'][] = 'ddmRxPowThresholdHighWarn';
$config['os']['jetstream']['oids']['no_bulk'][] = 'ddmTxPowThresholdLowAlarm';
$config['os']['jetstream']['oids']['no_bulk'][] = 'ddmTxPowThresholdLowWarn';
$config['os']['jetstream']['oids']['no_bulk'][] = 'ddmTxPowThresholdHighAlarm';
$config['os']['jetstream']['oids']['no_bulk'][] = 'ddmTxPowThresholdHighWarn';
$config['os']['jetstream']['oids']['no_bulk'][] = 'ddmStatusEntry';
